frontend/public/components/machine-set.tsx

```diff
@@ -276,6 +317,19 @@ export const MachineSetList: React.SFC = (props) => {
   };
 
   const MachineSetTableRow: RowFunction<MachineSetKind> = ({ obj, index, key, style }) => {
+    const relatedMachines =
+      isMachineLoaded &&
+      machines.filter(
+        (machine) =>
+          machine.metadata.labels['machine.openshift.io/cluster-api-machineset'] ===
+          obj.metadata.name,
+      );
+    const relatedNodes =
+      isNodeLoaded && relatedMachines.length > 0
+        ? nodes.filter((node) =>
+            relatedMachines.some((machine) => node.metadata.uid === machine.status.nodeRef.uid),
```

Comment:
I would add optional chaining here to be safe.

```suggestion
            relatedMachines.some((machine) => node.metadata.uid === machine.status?.nodeRef?.uid),
```